### DISCUSSION TITLE
Add ability to refresh skill calculator for current skill

### DISCUSSION THREAD

**Discussion by iitalics** [runelite/runelite#17120] [Ideas]
I noticed that in 84a23cb59d94c0382d777e9af430aa87f4b434cf the functionality was reverted where clicking the skill icon that is already selected would reset the current/target level/xp fields. Understandably there was a downside to the old behavior, since it reset the target, but it was very useful for resetting the current xp. Picture the following scenario:

1. I set up the skill calculator to see how many crafting actions of some combination (e.g. create glass + blow lantern lens) I need for my goal.
2. I perform some actions in the meantime, increasing my current xp.
3. The skill calculator still displays my old xp.
4. I want to update the skill calculator to see how many more actions I need to level up.

The only options available to me are:

- Manually check my current xp value and type that number into the field. This is cumbersome because the number may be 7-8 digits. You cannot even use the clipboard to compensate.
- Deselect crafting and then reselect it to update the current xp value. This also deselects my combination of actions, so I have to scroll down the list and find them again.

Clearly neither of these are very good. There should be some function exposed in the UI for updating the current xp field to the actual current xp.

**Comment by Felanbird** [comment]
This was discussed in the discord but the unintended change that broke this functionality earlier this year, was due to fixing an issue where clicking the hub plugin tab would cause immense lag due to the reloading. In comparison to the reloading of one exp value in the skill calculator tab.

As a note though, outside of the people who noticed this change immediately, I've seen no reports since July until your discussion today. So I'm unsure how popular this feature was.

**Reply by iitalics** [reply]
It's only a minor annoyance but I've definitely hit this a few times since the behavior changed. I was also watching an older youtube video of someone playing the game and I noticed them use the trick of clicking the already-selected skill to refresh the values. So if there is an easy way to bring back some of the old behavior, without reintroducing this lag problem, I think that would be very welcome.

**Comment by Operativekiwi** [comment]
I would also like this feature to be included! Bit annoying to have to click off the skill to refresh the current level/exp
